dask/array/tests/test_array_core.py

```diff
@@ -2643,19 +2643,24 @@ def assert_chunks_are_of_type(x):
     assert_chunks_are_of_type(dx[0:5][:, 0])
 
 
-def test_from_array_getitem():
+@pytest.mark.parametrize("wrap", [True, False])
+@pytest.mark.parametrize("inline_array", [True, False])
+def test_from_array_getitem(wrap, inline_array):
     x = np.arange(10)
+    called = False
 
-    def my_getitem(x, ind):
-        return x[ind]
-
-    y = da.from_array(x, chunks=(5,), getitem=my_getitem)
+    def my_getitem(a, ind):
+        nonlocal called
+        called = True
+        return a[ind]
 
-    for k, v in y.dask.items():
-        if isinstance(v, tuple):
-            assert v[0] is my_getitem
+    xx = MyArray(x) if wrap else x
+    y = da.from_array(xx, chunks=(5,), getitem=my_getitem, inline_array=inline_array)
 
     assert_eq(x, y)
+    # If we have a raw numpy array we eagerly slice, so custom getters
+    # are not called.
```

Comment:
I don't quite follow this part. Could you explain a bit more?

Reply:
There's a (somewhat old) special casing for numpy arrays here: https://github.com/dask/dask/blob/e07c98c670027008f693a9f44ce6157093b37b35/dask/array/core.py#L3310-L3316

This eagerly slices them rather than building a delayed graph (which is what we want for lazy array-likes).

I wonder if we still need this behavior in light of @ncclementi's work in #8015. I should check

Reply:
Okay: having thought about it a bit, I think #8015 is intended for a different enough use-case that this special-casing is still fine.